test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -1,230 +1,732 @@
+            Set-Content -Path $testFile -Value "Test Line 1`nTest Line 2`nTest Line 3`nTest Line 4`nTest Line 5`nTest Line 6"
+            $result = Get-Content -Path $testFile -Tail 2
+            $result[0] | Should Be "Test Line 5"
+            $result[1] | Should Be "Test Line 6"
+            $result[2] | Should BeNullOrEmpty
+        }
     }
 
-    It "Verify HOME" {
-        $homePath = $HOME
-        $tildePath = (resolve-path -Path ~).Path
-        $homePath | Should Be $tildePath
+    Context "Validate Resolve-Path Cmdlet Parameters" {
+        It "Verify LiteralPath" {
+            $result = Resolve-Path -LiteralPath "TestDrive:\"
+            ($result.Path.TrimEnd('/\')) | Should Be "TestDrive:"
+        }
     }
 
-    It "Verify relative" {
-        $relativePath = Resolve-Path -Path . -Relative
-        $relativePath | Should Be (Join-Path "." "")
+    Context "Validate Join-Path Cmdlet Parameters Remotely" {
+        BeforeAll {
+            New-Item -Path $testFile -ItemType File > $null
+        }
+
+        It "Validate Resolve" {
+            $result = Join-Path -Path . -ChildPath $testFile -Resolve
+            if ($IsWindows) {
+                $result | Should BeExactly "TestDrive:\$testFile"
+            }
+            else {
+                $result | Should BeExactly "TestDrive:/$testFile"
+            }
+        }
     }
-}
 
-Describe "Validate Basic Path Cmdlets" -Tags "CI" {
-    It "Verify Convert-Path" {
-        $result = Convert-Path "."
-        $result | Should Be (Get-Location).Path
+    Context "Validate Split-Path Cmdlet Parameters Remotely" {
+        BeforeAll {
+            New-Item -Path $testDir -ItemType Directory > $null
+            $fullPath = Resolve-Path $testDir
+        }
+
+        It "Validate Parent" {
+            $result = Split-Path -Path $fullPath -Parent -Resolve
+            ($result.TrimEnd('/\')) | Should Be "TestDrive:"
+        }
+
+        It "Validate IsAbsolute" {
+            $resolved = Split-Path -Path . -Resolve -IsAbsolute
+            $unresolved = Split-Path -Path . -IsAbsolute
+            $resolved | Should Be $true
+            $unresolved | Should Be $false
+        }
+
+        It "Validate Leaf" {
+            $result = Split-Path -Path $fullPath -Leaf
+            $result | Should Be $testDir
+        }
+
+        It "Validate NoQualifier" {
+            $result = Split-Path -Path $fullPath -NoQualifier
+            ($result.TrimStart('/\')) | Should Be $testDir
+        }
+
+        It "Validate Qualifier" {
+            $result = Split-Path -Path $fullPath -Qualifier
+            ($result.TrimEnd('/\')) | Should Be "TestDrive:"
+        }
     }
 
-    It "Verify Join-Path" {
-        $result = Join-Path -Path "TestDrive:" -ChildPath temp
+    Context "Validate Get-ChildItem Cmdlet Parameters Remotely" {
+        BeforeAll {
+            New-Item -Path $testFile -ItemType File > $null
+            New-Item -Path $testFile2 -ItemType File > $null
```

Comment:
Cleanup is not necessary because TestDrive:\ gets cleaned up at the end of every Context.